### PR TITLE
[s3 gateway] Handle all path characters in SigV2 and add Nessie S3-ish test

### PR DESCRIPTION
**Pull request by arielshaqed**
Use correctly-coded path when checking sigV2 signatures in the S3 gateway.

Use the minIO client to test S3 uploads and downloads with both V2 and V4
signatures.  (Unlike the AWS S3 client, the minIO client supports signing
using SigV2.)

### CODE REVIEW

File: nessie/s3_gateway_test.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+			for i := 0; i < parallelism; i++ {
+				wg.Add(1)
+				go func() {
+					client, err := minio.New(endpoint, &minio.Options{
+						Creds:  creds,
+						Secure: false,
+					})
+					if err != nil {
+						t.Error(fmt.Sprintf("minio.New: %s", err))
```
Comment (guy-har) [inline nessie/s3_gateway_test.go:66]: why aren't you returning in case of error? will be hard to continue without a client

**Review by guy-har — APPROVED** [review]
Looks good! 
We really needed these tests,
Thanks!

File: pkg/gateway/sig/v4.go [hunk 2]
```diff
@@ -277,6 +277,7 @@ func (ctx *verificationCtx) getAmzDate() (string, error) {
 	// parse signature date
 	sigTS, err := time.Parse(v4shortTimeFormat, ctx.AuthValue.Date)
 	if err != nil {
+		fmt.Printf("[DEBUG] failed to parse auth-date %s by %s: %s\n", ctx.AuthValue.Date, v4shortTimeFormat, err)
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply pkg/gateway/sig/v4.go:280]: :cry: sorry, thanks!

File: nessie/s3_gateway_test.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+						_, err := client.PutObject(
+							ctx, repo, o.Path, strings.NewReader(o.Content), int64(len(o.Content)), opts)
+						if err != nil {
+							t.Error(fmt.Sprintf("minio.Client.PutObject(%s): %s", o.Path, err))
+						}
+
+						download, err := client.GetObject(
+							ctx, repo, o.Path, minio.GetObjectOptions{})
+						if err != nil {
+							t.Error(fmt.Sprintf("minio.Client.GetObject(%s): %s", o.Path, err))
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply nessie/s3_gateway_test.go]: You're right, that was silly of me.

File: nessie/s3_gateway_test.go (hunk 1 above)
Comment (arielshaqed) [reply nessie/s3_gateway_test.go]: Good catch!  Dying fatally outside the goroutine when this fails.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Pull request merged**